meson.build

```diff
@@ -944,18 +944,36 @@ have_dns = (
 )
 
 if enable_zeroconf
-    if avahi.found()
-        have_zeroconf = true
-        cdata.set('USE_ZEROCONF', 1)
-        cdata.set('HAVE_AVAHI', 1)
-        cdata.set('freebsd_zeroconf', 'avahi_daemon')
-        zeroconf_provider += 'Avahi'
-    elif have_dns
-        have_zeroconf = true
-        cdata.set('USE_ZEROCONF', 1)
-        cdata.set('HAVE_MDNS', 1)
-        cdata.set('freebsd_zeroconf', 'mdnsd')
-        zeroconf_provider += 'mDNS'
+    if host_os == 'darwin'
+        # On Darwin/macOS, prefer mDNS (Bonjour) over Avahi
+        if have_dns
+            have_zeroconf = true
+            cdata.set('USE_ZEROCONF', 1)
+            cdata.set('HAVE_MDNS', 1)
+            cdata.set('freebsd_zeroconf', 'mdnsd')
```

Comment:
Minor remark: We don't need to set the freebsd string in the Darwin branch

Reply:
If I remove or change the line cdata.set('freebsd_zeroconf', 'mdnsd') the build always ends up using Avahi.

Reply:
Yes. I edited just those specific lines. Testing on 2 x macOS virtual machines. Tried removiing the line and just removing the frebsd part.
 cdata.set('mdnsd')